Source/Parallelization/WarpXRegrid.cpp

```diff
@@ -166,6 +166,45 @@ WarpX::RemakeLevel (int lev, Real /*time*/, const BoxArray& ba, const Distributi
             rho_fp[lev] = std::move(pmf);
         }
 
+#ifdef WARPX_USE_PSATD
+        if (maxwell_solver_id == MaxwellSolverAlgo::PSATD) {
+            if (spectral_solver_fp[lev] != nullptr) {
+                // Get the cell-centered box
+                BoxArray realspace_ba = ba;   // Copy box
```

Comment:
Could these lines, 172 through 186, be moved into the Alloc routine, since they are duplicated here and in WarpX.cpp? The same for the Cartesian version.

Reply:
Possibly, how would you deal with the difference between lines 1286--1288 vs. 1424 in WarpX.cpp (and similar for the Cartesian version)?

Reply:
Alright, I have moved some of those lines inside the functions.